packages/blade/src/components/Box/__tests__/styled-props.test.ts

```diff
@@ -0,0 +1,44 @@
+import { filterStyledProps, getStyledProps } from '~components/Box/styled-props';
+
+describe('getStyledProps', () => {
+  it('should ignore props that are not styled-props', () => {
+    expect(getStyledProps({ marginTop: 'spacing.2', padding: 'spacing.1' })).toMatchInlineSnapshot(`
```

Comment:
why did it truncate padding? because it doesn't has array prop and string literal?

Reply:
nope. because we don't support padding on styled-props. We only support padding on Box

Reply:
do we throw a warning so devs understand what exactly happened? otherwise we'll get 50 questions on slack for the same 😭

Reply:
The autocomplete itself won't show `padding` attribute on components no? It is equivalent to now having that prop. But yes good point. Let me add this in documentation so we can just link documentation 😆

Reply:
Oh actually Divyanshu has explained it in RFC. We can link to this if someone asks https://github.com/razorpay/blade/blob/master/rfcs/2023-01-06-layout.md#layout-based-styling-props